slow_benchmarks/matrices.py

```diff
@@ -0,0 +1,455 @@
+            [     1/4 + 13*z/4,    -825/64 - 147*I/32,          21/8 + x,    -537/64 + 143*y/16,    -5/8 - 39*z/16,   2473/256 + 137*I/64,    -149/64 + 49*x/32,   -177/128 - 1369*y/128]]'''))
+
+_n         = 6
+A, B, C, D = A[:_n,:_n], B[:_n,:_n], C[:_n,:_n], D[:_n,:_n]
+
+# has eigenvects
+AE = Matrix([
+    [    0, 1 - x, x + 1, 1 - x],
+    [1 - x, x + 1,     0, x + 1],
+    [    0, 1 - x, x + 1, 1 - x],
+    [    0,     0,     1 - x, 0]])
+BE = Matrix([
+    [    x,  x + 1,      0,      0],
+    [x + 8,  x + 9,      0, x + 11],
+    [    0, x + 17, x + 18,      0],
+    [    0,      0,      0, x + 27]])
+CE = Matrix(S('''[
+            [             -3/4,                     0,         1/4 + I/2,                     0],
+            [                0, -177/128 - 1369*I/128,                 0, -2063/256 + 541*I/128],
+            [          1/2 - I,                     0,                 0,                     0],
+            [                0,                     0,                 0, -177/128 - 1369*I/128]]'''))
+DE = Matrix(S('''[
+            [                0,       45/32 - 37*I/16,         1/4 + x/2,                     0],
+            [                0,                     0,                 0, -2063/256 + 541*y/128],
+            [                0,                     0,                 0,                     0],
+            [                0,   2473/256 + 137*z/64,                 0,                     0]]'''))
+
+# diagonalizable
+AD = Matrix([
+    [x + 1,  1 - x,      0,      0],
+    [1 - x,  x + 1,      0,  x + 1],
+    [    0,  1 - x,  x + 1,      0],
+    [    0,      0,      0,  x + 1]])
+BD = Matrix([
+    [    x,  x + 1,      0,      0],
+    [x + 8,  x + 9,      0, x + 11],
+    [    0, x + 17, x + 18,      0],
+    [    0,      0,      0, x + 27]])
+CD = Matrix(S('''[
+    [             -3/4,       45/32 - 37*I/16,                   0,                     0],
+    [-149/64 + 49*I/32, -177/128 - 1369*I/128,                   0, -2063/256 + 541*I/128],
+    [                0,         9/4 + 55*I/16, 2473/256 + 137*I/64,                     0],
+    [                0,                     0,                   0, -177/128 - 1369*I/128]]'''))
+DD = Matrix(S('''[
+    [             -3/4,       45/32 - 37*I/16,                   0,                     0],
+    [-149/64 + 49*z/32, -177/128 - 1369*I/128,                   0,  541*y/128 - 2063/256],
+    [                0,         9/4 + 55*I/16, 137*z/64 + 2473/256,                     0],
+    [                0,                     0,                   0, -1369*x/128 - 177/128]]'''))
+
+ADS = SparseMatrix(AD)
+BDS = SparseMatrix(BD)
+CDS = SparseMatrix(CD)
+DDS = SparseMatrix(DD)
+
+O4 = Matrix(4, 1, [1, 1, 1, 1])
+
+
+class TimePow4:
+    def time_A(self): A**4
+    def time_B(self): B**4
+    def time_C(self): C**4
+    def time_D(self): D**4

```

Comment:
Maybe could do
```
class _TimeMatrixBase:
    def time_A(self): self.func(A)
    def time_B(self): self.func(B)
    def time_C(self): self.func(C)
    def time_D(self): self.func(D)

class TimePow4(_TimeMatrixBase):
    def func(self, M):
        return M**4
```

Reply:
Probably, but at this point with my potentially limited time I am focusing on the actual matrix code...